jpa/org.eclipse.persistence.jpa/src/org/eclipse/persistence/internal/jpa/EntityManagerImpl.java

```diff
@@ -2779,13 +2781,33 @@ public LockModeType getLockMode(Object entity) {
             } else if (cls.equals(SessionBroker.class)) {            
                 return (T) this.getSessionBroker();
             } else if (cls.equals(java.sql.Connection.class)) {
-                UnitOfWorkImpl unitOfWork = (UnitOfWorkImpl) this.getUnitOfWork();
-                if(unitOfWork.isInTransaction() || unitOfWork.getParent().isExclusiveIsolatedClientSession()) {
-                    return (T) unitOfWork.getAccessor().getConnection();
+                final UnitOfWorkImpl unitOfWork = (UnitOfWorkImpl) this.getUnitOfWork();
+                final Accessor accessor = unitOfWork.getAccessor();
+                if (unitOfWork.getParent().isExclusiveIsolatedClientSession()) {
+                    // If the ExclusiveIsolatedClientSession hasn't serviced a query prior to the unwrap, 
+                    // there will be no available Connection.
```

Comment:
Didn't we establish a connection during DatabaseSessionImpl.loginAndDetectDatasource()? What happens with that connection?

Reply:
The idea with an Exclusive Isolated Client Session is to have a unique connection that is not shared between EntityManagers during use -- it is often used for enhanced security per connection (something you probably don't want to share with a different EntityManager that you intend to have a different security context with).  So the connection from DatabaseSessionImpl.loginAndDetectDatasource() is not really relevant.